### PR TITLE
Use AST parser to validate alt text - markdown-lint

### LINKED ISSUE #30: Ignore images in code snippets
If a user adds an image in a code snippet, we should ignore the alt text.

```
<img alt='' src='' /> 
```

### PR DESCRIPTION
**Pull request by kendallgassner**
### What

Closes #30 

I updated the alt-text-bot to use an AST (markdown-lint) to parse images for bad alt text. Using an AST allows us to ignore any images defined in code blocks.

### Why is this cool
- In the future, we can add additional linting rules
- The workflow is easier to read in javascript
- We can test more easily
- We can ensure that there aren't false positives 
- We can provide more feedback. For example, it is easier to indicate what line the alt text is found on.

### CODE REVIEW

File: src/index.test.js [hunk 1]
```diff
@@ -0,0 +1,128 @@
+import { validate } from "./validate.js";
+
+test("no-alt-text: should return errors", async () => {
+  let result = await validate("![]()");
+  expect(result[0]).toBe("Images should have alternate text (alt text)");
+  // expect(result[0]).toBe("Images should have alternate text (alt text)");
```
Comment (kendallgassner) [inline src/index.test.js:6]: Commented out until `no-alt-text ` is updated to return error for HTML images

File: action.yml [hunk 2]
```diff
@@ -1,78 +1,79 @@
...
+  icon: "eye"
+  color: "purple"
 runs:
   using: "composite"
   steps:
     - name: Runs alt text check and adds comment
       run: |
-          source ${{ github.action_path }}/flag-alt-text.sh
-          source ${{ github.action_path }}/queries.sh
+        source ${{ github.action_path }}/flag-alt-text.sh
```
Comment (khiga8) [inline action.yml:11]: Looks like we can delete this given the file has been removed!

File: src/index.test.js [hunk 3]
```diff
@@ -0,0 +1,127 @@
+import { validate } from "./validate.js";
```
Comment (khiga8) [inline src/index.test.js:1]: Beautiful test coverage!

File: action.yml (hunk 2 above)
Comment (khiga8) [reply action.yml:11]: I am testing this branch in one of my repos! ([Related workflow error](https://github.com/khiga8/link-checker/actions/runs/6487639086/job/17618370808))

File: action.yml [hunk 4]
```diff
@@ -1,78 +1,79 @@
...
+
+        if [[ $flag = true ]]; then
+          if [[ $type = pr_comment ]] || [[ $type = pr_description ]]; then
+            gh pr comment $issue_url --body "$message"
+          elif [[ $type = issue_comment ]] || [[ $type = issue_description ]]; then
+            gh issue comment $issue_url --body "$message"
+          elif [[ $type = discussion_description ]]; then
+            addDiscussionComment $discussion_node_id "$message"
+          elif [[ $type = discussion_comment ]]; then
+            addDiscussionComment $discussion_node_id "$message" $reply_to_id
```
Comment (khiga8) [inline action.yml:74]: I would love if we can eventually move all of this bash logic into a well-tested node script! Out of scope here though :)

File: src/index.js [hunk 5]
```diff
@@ -0,0 +1,9 @@
+import { validate } from "./validate.js";
+
+const [content] = process.argv.slice(2);
+
+const run = async () => {
+  console.log((await validate(content)).length);
```
Comment (khiga8) [inline src/index.js:6]: If this is 0, would the presence of the log trigger an error in the bash line above?

```
flag=$(node index.js "$content")
```

Comment (kendallgassner) [reply src/index.js:6]: Oh maybe I need to be more explicit 🤔 

```suggestion
  console.log((await validate(content)).length > 0);
```

File: action.yml (hunk 2 above)
Comment (kendallgassner) [inline action.yml:11]:
```suggestion
```

File: action.yml (hunk 4 above)
Comment (kendallgassner) [reply action.yml:73]: Agreed!!

File: action.yml [hunk 6]
```diff
@@ -1,78 +1,78 @@
...
+          elif [ ${{ github.event.discussion }} ]; then
+            type=discussion_description
+            content=$DISCUSSION_BODY
+            discussion_node_id='${{ github.event.discussion.node_id }}'
+            user=${{ github.event.discussion.user.login }}
+            target=" your discussion body"
           fi
-          message="Uh oh! @$user, the image you shared is missing helpful alt text. Check $target.
+        fi
+        npm ci
```
Comment (kendallgassner) [inline action.yml:54]:
```suggestion
        npm i
```

**Comment by kendallgassner** [comment]
Proof of concept run: https://github.com/khiga8/link-checker/issues/9

@khiga8 ... i had to check in node_modules..... but it works

**Comment by khiga8** [comment]
Eeep sorry I didn't get to review this today, I'll take a look first thing tomorrow!

**Review by khiga8 — APPROVED** [review]
Great work!

**Comment by kendallgassner** [comment]
[Checking in node_modules as advised on github action documentation](https://docs.github.com/en/actions/creating-actions/creating-a-javascript-action#commit-tag-and-push-your-action-to-github).

**Pull request merged**